### PR TITLE
refactor: add --upgrade flag in front of pip install instances

### LINKED ISSUE #553: Add --upgrade flag in front of all the pip install instances
### 🐛 Describe the bug

As we are rapidly developing our package, we want our users to be on the latest version so that they can avail all the functionalities.
For this we want to recommend to use `--upgrade` package when installing.

One such instance in readme.

```
pip install --upgrade embedchain
```

We want to update this everywhere. Example
* replace
```
pip install "embedchain[poe]"
```
with
```
pip install --upgrade "embedchain[poe]"
```

### PR DESCRIPTION
**Pull request by ShameGod**
## Description

Add --upgrade flag in front of pip install instances
Fixes #553 

## Type of change

- [x] Refactor (does not change functionality, e.g. code style improvements, linting)
- [x] Documentation update

## How Has This Been Tested?

- [x] Unit Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by cachho** [comment]
if you want you can integrate #541 as well. thanks for your contribution.

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Book a [1-on-1 Session](https://cal.com/taranjeetio/ec) with Taranjeet, the foun
 ## 🔧 Quick install
 
 ```bash
-pip install --upgrade embedchain
+pip install --upgrade "embedchain"
```
Comment (taranjeet) [inline README.md:18]: Hey thanks for your contribution.
But we need double quotes where there is extra

eg: (where double quotes are needed)
```
pip install --upgrade "embedchain[poe]"
```

eg: (where double quotes are not needed)
```
pip install --upgrade embedchain
```

File: docs/examples/whatsapp_bot.mdx [hunk 2]
```diff
@@ -7,7 +7,7 @@ title: '💬 WhatsApp Bot'
 1. Install embedchain python package:
 
 ```bash
-pip install embedchain
+pip install --upgrade "embedchain"
```
Comment (deshraj) [inline docs/examples/whatsapp_bot.mdx:10]:
```suggestion
pip install --upgrade embedchain
```

File: README.md (hunk 1 above)
Comment (deshraj) [inline README.md:18]:
```suggestion
pip install --upgrade embedchain
```

File: docs/quickstart.mdx [hunk 3]
```diff
@@ -6,7 +6,7 @@ description: '💡 Start building LLM powered bots under 30 seconds'
 Install embedchain python package:
 
 ```bash
-pip install embedchain
+pip install --upgrade "embedchain"
```
Comment (deshraj) [inline docs/quickstart.mdx:9]:
```suggestion
pip install --upgrade embedchain
```

**Review by deshraj — APPROVED** [review]

**Comment by deshraj** [comment]
@ShameGod thanks for the contribution. The PR looks good but can you please do resolve the conflicts?

**Comment by taranjeet** [comment]
@ShameGod : let me know if you are still interested in working on this issue?

**Comment by taranjeet** [comment]
Thanks for the contribution on this.
Closing this as the support is already live.

**Pull request closed without merging**